### PR TITLE
CRM:4287 - add searchPrimaryDetailsOnly setting

### PR DESCRIPTION
**Pull request by jitendrapurohit**
https://issues.civicrm.org/jira/browse/CRM-4287

### CODE REVIEW

File: CRM/Contact/BAO/Query.php [hunk 1]
```diff
@@ -514,7 +514,7 @@ public function initialize($apiEntity = NULL) {
       CRM_Financial_BAO_FinancialType::buildPermissionedClause($this->_whereClause, $component);
     }
 
-    $this->_fromClause = self::fromClause($this->_tables, NULL, NULL, $this->_primaryLocation, $this->_mode);
+    $this->_fromClause = self::fromClause($this->_tables, NULL, NULL, $this->_primaryLocation, $this->_mode, $apiEntity);
```
Comment (eileenmcnaughton) [inline CRM/Contact/BAO/Query.php]: when I last looked at this apiEntity seemed to duplicate $this->_mode.

**Comment by monishdeb** [comment]
Tested working fine. 

1. Created two email ```test1@test.com``` (as primary) and ```test2@test.com``` for contact A 
image

2. On ```searchPrimaryDetailsOnly = TRUE``` AND ```email LIKE %test2%``` 
image

3.  On ```searchPrimaryDetailsOnly = FALSE``` AND ```email LIKE %test2%``` 
image

**Comment by monishdeb** [comment]
Also works for phone, im and openid

File: CRM/Contact/BAO/Query.php (hunk 1 above)
Comment (monishdeb) [inline CRM/Contact/BAO/Query.php]: Actually $this->_mode contain the constant values specified in https://github.com/jitendrapurohit/civicrm-core/blob/master/CRM/Contact/BAO/Query.php#L47  and ```apiEntity``` is assigned with string entity values e.g. Contact, Participant etc.

Comment (eileenmcnaughton) [inline CRM/Contact/BAO/Query.php]: Just for the record, I still don't understand how that isn't just duplicating mode in a different format (you don't need to respond to this - it has been merged & it is not a 'live' concern)

**Comment by petednz** [comment]
Does this work for QuickSearch - or will that require additional work? @jitendrapurohit an estimate if that is the case pls

**Pull request merged**